### PR TITLE
boards: fysetc: add FYSETC UCAN USB to CAN 2.0B adapter board

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Add support for the open-source FYSETC UCAN USB to CAN 2.0B board.

Documentation preview: https://builds.zephyrproject.io/zephyr/pr/80796/docs/boards/fysetc/ucan/doc/index.html

### CODE REVIEW

**Comment by henrikbrixandersen** [comment]
Yet another board well-suited for running https://github.com/CANnectivity/cannectivity

File: boards/fysetc/ucan/ucan.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+identifier: ucan
+name: FYSETC UCAN
+type: mcu
+arch: arm
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
```
Comment (kartben) [inline boards/fysetc/ucan/ucan.yaml:8]:
```suggestion
```

https://github.com/zephyrproject-rtos/zephyr/pull/80466

**Review by kartben — CHANGES_REQUESTED** [review]
Awesome!

File: boards/fysetc/ucan/ucan.yaml (hunk 1 above)
Comment (henrikbrixandersen) [reply boards/fysetc/ucan/ucan.yaml]: Thank you for reviewing! I've removed `xtools` from the list.

**Review by kartben — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by henrikbrixandersen** [comment]
Added support for using `west flash` with the `dfu-util` flasher.

File: boards/fysetc/ucan/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+The STM32F072CB PLL is driven by an external crystal oscillator (HSE) running at 8 MHz and
+configured to provide a system clock of 48 MHz.
+
+Programming and Debugging
+*************************
+
+Build and flash applications as usual (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+If flashing via USB DFU, short pins ``B0`` and ``3V3`` when applying power to the candleLight in
```
Comment (kartben) [inline boards/fysetc/ucan/doc/index.rst:54]: not candlelight :)

Comment (henrikbrixandersen) [reply boards/fysetc/ucan/doc/index.rst]: Gah, thanks! You see where I got my inspiration from.

**Review by kartben — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**